docs: update E2E pipeline process in developer docs (#642)
Azure/aks-engine: docs/community/developer-guide.md

The following steps constitute the AKS Engine CI pipeline:

1. Contributor opens a Pull Request (PR) against the AKS Engine project

> 1. A member of the AKS Engine team comments on the PR to trigger an Azure DevOps job that

Reviewer: My most nit comment till date:
```suggestion
1. An AKS Engine team member comments on the PR to trigger an Azure DevOps job that
```

Reviewer: ```suggestion
2. A member of the AKS Engine team comments on the PR to trigger an Azure DevOps job that
```

> 1. A member of the AKS Engine team comments on the PR to trigger an Azure DevOps job that

Reviewer: ```suggestion
2. An AKS Engine team member comments on the PR to trigger an Azure DevOps job that
```
